homeassistant/components/roborock/config_flow.py

```diff
@@ -153,3 +163,104 @@ def _create_entry(
+        return RoborockOptionsFlowHandler(config_entry)
+
+
+class RoborockOptionsFlowHandler(config_entries.OptionsFlowWithConfigEntry):
+    """Handle an option flow for Roborock."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        return self.async_show_menu(step_id="init", menu_options=[DOMAIN, MAPS])
+
+    async def async_step_roborock(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options domain wide."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title="", data={**self.config_entry.options, **user_input}
+            )
+        return self.async_show_form(
+            step_id=DOMAIN,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_INCLUDE_SHARED,
+                        default=self.config_entry.options.get(
+                            CONF_INCLUDE_SHARED, DEFAULT_INCLUDE_SHARED
+                        ),
+                    ): bool
+                }
+            ),
+        )
+
+    async def async_step_maps(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Open the menu for the map options."""
+        return self.async_show_menu(step_id=MAPS, menu_options=[DRAWABLES, SIZES])
+
+    async def async_step_sizes(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the map object size options."""
+        if user_input is not None:
+            new_sizes = {
+                SIZES: {**self.config_entry.options.get(SIZES, {}), **user_input}
+            }
+            return self.async_create_entry(
+                title="", data={**self.config_entry.options, **new_sizes}
+            )
+        data_schema = {}
+        for size, default_value in DEFAULT_SIZES.items():
+            data_schema[
+                vol.Required(
+                    size.value,
+                    default=self.config_entry.options.get(SIZES, {}).get(
+                        size, default_value
+                    ),
+                )
+            ] = vol.All(vol.Coerce(float), vol.Range(min=0))
+        return self.async_show_form(
+            step_id=SIZES,
+            data_schema=vol.Schema(data_schema),
+        )
+
+    async def async_step_drawables(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the map object drawable options."""
+        if user_input is not None:
+            new_drawables = {
+                DRAWABLES: {
+                    **self.config_entry.options.get(DRAWABLES, {}),
+                    **user_input,
+                }
+            }
+            return self.async_create_entry(
+                title="", data={**self.config_entry.options, **new_drawables}
+            )
```

Comment:
```suggestion
            self.options.setdefault(DRAWABLES, {}).update(user_input)
            return self.async_create_entry(
                title="", data=self.options
            )
```

Please update also other places too